app/js/components/tabs/tabcomponents/patientComponent.jsx

```diff
@@ -49,12 +49,23 @@ class PatientComponent extends Component {
 
   searchDemographics(event) {
     event.preventDefault();
-    const { gender, minAge, maxAge, livingStatus } = this.state;
-    let { startDate, endDate } = this.state;
+    const { gender, livingStatus } = this.state;
+    let { startDate, endDate, minAge, maxAge } = this.state;
     const searchParameters = { gender };
     // add appropriate age constraints to the search parameters
     if (minAge || maxAge) {
       if (minAge && maxAge) {
+        // coerce the values of minAge and maxAge to numbers for comparison    
+        minAge = Number(minAge);
+        maxAge = Number(maxAge);
+
+        // switch the min and max ages if the min age is greater than the max age
+        if (minAge > maxAge) {
+          const minAgeStore = minAge;
```

Comment:
Hi @kingisaac95 don't you think  it would be better to prevent the user from inputing a minAge that is higher than the maxAge from the front-end, as opposed to having an abstracted logic that alters the user search query in the background. :-).